### DISCUSSION TITLE
How to create a github repository with submodules and hidden files and directories

### DISCUSSION THREAD

**Discussion by jschwartzman** [community/community#45957] [Actions]
### Select Topic Area

General

### Body

Please tell me how to publish a repository that contains submodules, hidden files and hidden directories. I have tried:
     git clone --recurse-submodules add --all https://github.com 
           /jschwartzman/x86_64_asm_tutorial.git .

What should this command be?
What would be the command to clone the repository (with all of its submodules and hidden files and directories once it is created?

Thank you,
John Schwartzman

**Comment by jge162** [comment]
`git clone --recurse-submodules --quiet https://github.com/<username>/<repository-name>.git`
`ls -al`
`git push origin master`

**Reply by jschwartzman** [reply]
git clone ... worked ok. It seems to have created an x86_64_asm_tutorial directory on my local machine. The directory does not have any .git or .vscode directories in the submodules.
Do I need to be in the new directory before I issue git push origin main?
cd x86_64_asm_tutorial
git push origin main
remote: Support for password authentication was removed on August 13, 2021. 
It seems I have an authentication issue. What's the best way to proceed?

**Reply by jge162** [reply]
you need to either generate an SSH key and add it to your GitHub account, or generate a personal access token and use it in place of your password when pushing changes.
